clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -1815,13 +1815,35 @@ class CIRGlobalOpLowering
     }
 
     // Rewrite op.
-    rewriter.replaceOpWithNewOp<mlir::LLVM::GlobalOp>(
+    auto llvmGlobalOp = rewriter.replaceOpWithNewOp<mlir::LLVM::GlobalOp>(
         op, llvmType, isConst, linkage, symbol, init.value(),
         /*alignment*/ 0, /*addrSpace*/ 0,
         /*dsoLocal*/ false, /*threadLocal*/ (bool)op.getTlsModelAttr(),
         /*comdat*/ mlir::SymbolRefAttr(), attributes);
+    auto mod = op->getParentOfType<mlir::ModuleOp>();
+    if (op.getComdat())
+      addComdat(llvmGlobalOp, comdatOp, rewriter, mod);
     return mlir::success();
   }
+
+private:
+  mutable mlir::LLVM::ComdatOp comdatOp = nullptr;
```

Comment:
simplify it after reading more about how LLVM uses ComdatOp. It looks to me that only one ComdatOp is needed for module for a given purpose, like here for all global variables. 
Then different SelectionKind can have different ComdatSelectorOp. And different globals can choose to have attribute with a given ComdatSelectorOp. 
So, really it makes no sense to keep a map of ComdatOp.